### DISCUSSION TITLE
Is there a way to detect if a scrollbar is scrolling?

### DISCUSSION THREAD

**Discussion by RadonCoding** [emilk/egui#2079] [Q&A]
The title pretty much says it all.

**Comment by RadonCoding** [comment]
I'm trying to make it so the scrollbar would be able to move infinitely without being locked to the end of the textbox.

**Reply by BrandonXZ** [reply]
not sure whether the infinite scrollbar is possible but you can access how many points the user has scrolled with the scroll_delta property of the input state struct... [https://docs.rs/egui/latest/src/egui/input_state.rs.html](url) . 
see line 36

**Reply by RadonCoding** [reply]
Basically what i want to do is to have a file and every time the user scrolls it will read at a certain offset in the file and display it. The thing that's stopping me though is that the scroll bar locks at the bottom of the textbox and i use the y offset in the scrollbar state as the file offset.

**Reply by BrandonXZ** [reply]
Sounds like you'll run into an issue with this being rendered 60x a second the larger the file gets due to the computation from saving the users scroll delta and calculating the right parts of the file based on what's been read/not read and what the size of the windows when you go to add the "reader"/"highlight" text component but may be possible by reading the file to string and calling [lines()](https://doc.rust-lang.org/std/io/struct.Lines.html) to break it into manageable chunks which you can limit to a certain width then map those lines to the length of the "viewable" area using components from [https://docs.rs/egui/latest/src/egui/widget_text.rs.html](url)

**Reply by RadonCoding** [reply]
reading the file into a string is not possible since it's a 18 GB file and i have 16 GB of RAM on my machine

**Reply by BrandonXZ** [reply]
https://github.com/emilk/egui#disadvantages-of-immediate-mode
